src/main/java/lesson25/MateRest.java

```diff
@@ -0,0 +1,41 @@
+package lesson25;
+
+import lesson10.xmljson.MateGroup;
+import lesson10.xmljson.Person;
+import lesson10.xmljson.Student;
+import org.glassfish.jersey.server.mvc.Viewable;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Path("/mate")
+public class MateRest {
+    private MateGroup mateGroup = MateGroup.mateGroupExampleCreator();
+
+    @GET
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.TEXT_HTML)
+    public Viewable index(@QueryParam("groupId") String groupId) {
+        Object o = new Object();
+        List<Object> some = (ArrayList) o;
+        if (groupId != null && groupId.equals("" + mateGroup.getId())) {
+            return new Viewable("/index", mateGroup);
+        }
+        return new Viewable("/index");
+    }
+
+    @GET
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.TEXT_HTML)
+    @Path("/students")
+    public Viewable stud(@QueryParam("yearOfBorn") String yearOfBorn) {
+        if (yearOfBorn != null && mateGroup.getStudents().stream().anyMatch(s -> yearOfBorn.equals("" + s.getYearOfBorn()))) {
+            Person student = mateGroup.getStudents().stream().filter(s -> yearOfBorn.equals("" + s.getYearOfBorn())).collect(Collectors.toList()).get(0);
+            return new Viewable("/stud-resp", student);
+        }
+        return new Viewable("/stud");
```

Comment:
Maybe here you need return new Viewable("/stud-resp"); ??

Reply:
na-ah